### DISCUSSION TITLE
New linter rule: no result destructuring

### DISCUSSION THREAD

**Discussion by Faithfinder** [TanStack/query#6629] [Ideas] [answered]
Shouldn't be part of the recommended set, but these days I find that `someQuery.isLoading` is easier to read (especially when there's multiple queries, but even with a single). My team would like to enforce the convention in the codebase, and I think the official plugin is a good place for such a rule.

**Comment by TkDodo** [comment, marked as answer]
This is stylistic at best so we're not going to include that. There is a lint rule prefer-destructuring (https://eslint.org/docs/latest/rules/prefer-destructuring) so I think it shouldn't be too hard to implement the opposite for yourself if you want to.
